### PR TITLE
xray-encrypt-key - error handling for key not found, don't assume len(resources) == 1

### PR DESCRIPTION
**Pull request by thisisshi**
Resolves: https://github.com/capitalone/cloud-custodian/issues/3052

```yaml
policies:
  - name: bad-alias-xray-key
    resource: account
    filters:
      - type: xray-encrypt-key
        key: alias/that/doesnt/exist
```

```yaml
policies:
  - name: not-filter-breakage
    resource: account
    filters:
      - not: 
        # an account id that doesn't match your account
        - account_id: 12345
        - type: xray-encrypt-key
          key: alias/some/alias
```

### CODE REVIEW

File: c7n/resources/account.py [hunk 1]
```diff
@@ -1007,11 +1007,20 @@ class XrayEncrypted(Filter):
...
+        else:
+            return []
+
         k = self.data.get('key')
         if k not in ['default', 'kms']:
             kmsclient = self.manager.session_factory().client('kms')
-            keyid = kmsclient.describe_key(KeyId=k)['KeyMetadata']['Arn']
+            try:
+                keyid = kmsclient.describe_key(KeyId=k)['KeyMetadata']['Arn']
+            except kmsclient.exceptions.NotFoundException:
```
Comment (kapilt) [inline c7n/resources/account.py]: why shouldn't this raise as an error?, its a policy misconfiguration.

**Comment by kapilt** [comment]
pushed an update to address the underlying issue wrt to and/or filters not invoking filters with empty resource sets, ci currently barfing on unrelated azure subscription errors.

**Comment by stefangordon** [comment]
> pushed an update to address the underlying issue wrt to and/or filters not invoking filters with empty resource sets, ci currently barfing on unrelated azure subscription errors.

Looking at this now

**Comment by kapilt** [comment]
@stefangordon thanks for looking!

**Pull request merged**